### DISCUSSION TITLE
calcMeanOrientation Changing Orientation After Merging Grains

### DISCUSSION THREAD

**Discussion by Savillian** [mtex-toolbox/mtex#1223] [Ask Anything]
Hi everyone,

After identifying twin boundaries and merging grains in one of my EBSD maps, I've found the 'calcMeanOrientation' flag for the 'merge' command changes some grain orientations significantly after the operation.

An example of this can be seen below. Prior to merging, the twin calculation process seems to give the boundaries around the twins their own orientation. And after merging, these orientations seem to take on the mean orientation of the grain they were housed in, but not part of the data. Not employing the 'calcMeanOrientation' flag will return grains without any orientations, so I haven't been able to find a workaround to this issue.

**Prior Microstructure to Merging (Note the dark blue around the twins on the left and the grey around the twins in the top grain)**
![050Large25GrainFinalMTEXTroubleshootSmoothedTwinInterfacesWithInterfaceOrientations](https://user-images.githubusercontent.com/38051659/134452234-c3daa00f-bff6-42f4-9317-21783084a46a.png)

**Twin Identification**
![050Large25GrainFinalMTEXTroubleshootBetaTwinBoundaries](https://user-images.githubusercontent.com/38051659/134452308-393088f7-223b-4672-b36c-279feaa4cefd.png)

**Merged Grains - Note the change in the non-twinned grain orientations**
![050Large25GrainBetaRemovedTwins](https://user-images.githubusercontent.com/38051659/134452350-d01c0100-2302-4cb0-bcd6-6cd327b68eec.png)

The orientations themselves aren't overly different from that of the original parent grain, but different enough on the IPF coloring scheme to be of concern. Does anyone know of a way around this or a filtering process I could use to bypass the issue?

**Script for generation included below:**
```
%% Grow Beta
ebsdSub = ebsd(('Titanium (Beta)'));
[grainsSub,ebsdSub.grainId] = calcGrains(ebsdSub,'angle',2*degree);
grainsSub = smooth(grainsSub,15);

F = halfQuadraticFilter;
F.alpha = 0.25;

% interpolate the missing data
ebsdSub_filled = smooth(ebsdSub,F,'fill',grainsSub);
ebsdSub_filled = ebsdSub_filled('indexed');

% Recalculate boundaries
[grainsSub_filled,ebsdSub_filled.grainId,ebsdSub_filled.mis2mean] = calcGrains(ebsdSub_filled,'angle',10*degree,'boundary','tight');
grainsSub_filled = grainsSub_filled.smooth(15)

grainsMean = grainsSub_filled.meanOrientation
[grainsMean,ebsdSub_filled.grainId] = calcGrains(ebsdSub_filled,'angle',10*degree);
grainsMean = smooth(grainsMean,15);

% extract all Ti-Ti grain boundaries
gB = grainsMean.boundary
gBTiTi = grainsMean.boundary('Titanium (Beta)','Titanium (Beta)');

figure
plot(grainsMean,grainsMean.meanOrientation)

figure
plot(gBTiTi,gBTiTi.misorientation.angle./degree,'linewidth',2)
mtexColorbar

ind = gBTiTi.misorientation.angle>49*degree & gBTiTi.misorientation.angle<51*degree;
mori = gBTiTi.misorientation(ind);

mori_mean = mean(mori,'robust')
round2Miller(mori_mean)

twinning = orientation.map(Miller(3,-1,1,CSBeta),Miller(3,1,-1,CSBeta),...
  Miller(0,1,1,CSBeta),Miller(0,1,1,CSBeta));

% and check which of them are twinning boundaries with threshold 5 degree
isTwinning = angle(gBTiTi.misorientation,twinning) < 5*degree;
twinBoundary = gBTiTi(isTwinning);

round(twinning.axis)

% the rotational angle
RotAngle = twinning.angle / degree;
% 50.5deg 110 axis Sigma 11 coincident site lattice

% Overlay Twins
figure
colorb2 = ipfKeyb.orientation2color(ebsdSub_filled('Titanium (Beta)').orientations);
plot(ebsdSub_filled('Titanium (Beta)'),colorb2)

hold on
plot(grainsMean.boundary,'lineWidth',0.5) % Adds boundaries to plot.
hold off

% plot the twinning boundaries
hold on
plot(twinBoundary,'linecolor','w','linewidth',4,'displayName','twin boundary')
hold off

%Merge Grains
[mergedGrains,parentId] = merge(grainsMean,twinBoundary,'calcMeanOrientation');
SEBSD = smooth(ebsdSub_filled,F,'fill',mergedGrains);

%% Plot Microstructure Without Twin Boundaries Highlighted
figure
colorb4 = ipfKeyb.orientation2color(SEBSD('Titanium (Beta)').orientations);
plot(SEBSD('Titanium (Beta)'),colorb4)
saveas(gcf,[Prfx,'SmoothedTwinInterfacesWithInterfaceOrientations.png'])

% Plot Merged Grains
figure
plot(mergedGrains,mergedGrains.meanOrientation)

hold on
plot(mergedGrains.boundary,'lineWidth',0.5) % Adds boundaries to plot.
hold off
```

Cheers,
Alec

**Comment by kilir** [comment]
Hi,
what would you like a single orientation of a merged grain to be? Obviously the mean is non-nonsensical in many cases. `merge()` also allows to supply a function handle, so in case one knows what should be computed, `merge()` can probably do it.
Hope that helps.
Cheers, Rüdiger
